### PR TITLE
feat: add `wet entity rename` command

### PR DESCRIPTION
**Pull request by petr-muller**
## Summary
- Adds `wet entity rename <old-name> <new-name>` to rename an entity.
- Since entity references are stored as literal bracket text in thought content and entity descriptions (not pure foreign keys), rename rewrites every stored reference to the old name — bare `[Old]` → `[New]`, and aliased `[Alias](Old)` → `[Alias](New)` (alias text preserved). `thought_entities` links are keyed by entity ID and are never touched.
- The whole operation (row rename + all rewrites) runs in a single transaction, so a failure can't leave things half-updated. Renaming onto an existing different entity is rejected (no merge); casing-only self-renames are allowed.
- See `specs/010-entity-rename.md` for full requirements/decisions.

## Test plan
- [x] `cargo nextest run` — 360/360 passed
- [x] `cargo clippy --all-targets` — no new warnings introduced
- [x] `cargo fmt --check` — clean
- [x] Manual smoke test: renamed an entity referenced both bare and via alias across two thoughts, confirmed both thoughts render the new name, confirmed no duplicate entity is created when re-editing an affected thought afterward

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/services/entity_parser.rs [hunk 1]
```diff
@@ -75,6 +75,53 @@ pub fn extract_unique_entities(text: &str) -> Vec<String> {
...
+
+    ENTITY_PATTERN
+        .replace_all(text, |caps: &Captures| {
+            let alias = caps[1].trim();
+            let target = caps.get(2).map(|m| m.as_str().trim());
+
+            match target {
+                Some(t) if !t.is_empty() => {
+                    if t.to_lowercase() == old_lower {
+                        format!("[{}]({})", alias, new_name)
```
Comment (petr-muller) [inline src/services/entity_parser.rs:113]: `new_name` is interpolated into bracket syntax unescaped, and there's no validation rejecting `[`, `]`, `(`, or `)` in `new_name` (entity creation implicitly excludes these via the parsing regex, but `entity rename`'s `new_name` comes straight from a CLI arg). Concretely: `wet entity rename sarah "Sarah]x"` turns a stored `[Sarah]` into `[Sarah]x]`, which `ENTITY_PATTERN` will then parse as entity `Sarah` followed by stray literal text `x]` on the next read — silently corrupting previously well-formed content. Consider rejecting `[`, `]`, `(`, `)` in `new_name` up front (same applies to the bare-rewrite branch at line 118).

**Review by petr-muller — COMMENTED** [review]
## Overview
Adds `wet entity rename <old-name> <new-name>`. Since entity references are stored as literal bracket text (not FK placeholders), rename rewrites bracket references in every entity description and every linked thought, then updates the entity row — all in one transaction. `thought_entities` links are untouched (keyed by ID). Well-specced in `specs/010-entity-rename.md`.

## Code quality
- Follows the existing `entity_edit.rs` command pattern closely (error messages, hints, structure) — good consistency.
- `rewrite_entity_references` reuses `ENTITY_PATTERN` and does plain string comparison rather than building a regex from user input, so there's no ReDoS/injection risk from entity names containing regex metacharacters.
- Transaction ordering is correct: descriptions + thought content are rewritten using the **old** name/row before the row itself is renamed, so lookups inside the loop stay valid.
- `ThoughtsRepository::update` is a raw content update with no entity re-linking side effect — the right choice here, since re-parsing on write would risk creating duplicate entities or dropping the just-renamed link.

## Correctness issue (see inline comment)
`rewrite_entity_references` writes `new_name` into bracket syntax unescaped. There's no validation preventing `new_name` from containing `[`, `]`, `(`, or `)`, and such a name can corrupt previously well-formed bracket references in stored content. See inline comment for a concrete repro.

## Minor observations
- The "new name already exists" collision check is done twice — once in `cli/entity_rename.rs::execute` before opening the transaction, and again inside `EntitiesRepository::rename`. Harmless for a single-user local CLI, just redundant.
- Like the pre-existing `entity_edit` command, `execute()` both `eprintln!`s a detailed error *and* returns `Err`, which `main.rs` prints again via `eprintln!("Error: {}", e)`. Pre-existing pattern, not introduced by this PR, but worth cleaning up at some point.
- README documents `wet entity edit` usage but not `wet entity rename` — minor doc gap.

## Test coverage
Strong — unit tests for `rewrite_entity_references` (bare/aliased/case-insensitive/coincidental-alias-text), repository tests for rename (success, casing-only self-rename, no-op, not-found, collision), integration tests covering link-by-ID preservation and rollback-on-failure, and contract tests through the CLI binary. Ran the new suite locally — all 10 new tests pass; `cargo clippy --all-targets` shows only pre-existing unrelated warnings.

## Risk
Low overall — additive and self-contained. Main risk is the unescaped-`new_name` bracket-corruption edge case above.

File: src/services/entity_parser.rs (hunk 1 above)
Comment (petr-muller) [reply src/services/entity_parser.rs:113]: Good catch — fixed in 87ebdce: `new_name` is now rejected up front if it contains `[`, `]`, `(`, or `)`, before any rewriting happens. Added a contract test covering all four characters plus verifying the original content is left untouched on rejection. Also documented the reserved-character constraint in the spec.

**Review by petr-muller — COMMENTED** [review]
Both items from the earlier review are addressed:

- `new-name` now rejects `[`, `]`, `(`, `)` (commit 87ebdce) before any DB lookup or transaction work — closes the bracket-corruption edge case flagged on `entity_parser.rs:113`. Covered by a new contract test (`test_entity_rename_rejects_bracket_characters_in_new_name`) exercising four bad-name variants and confirming rejected attempts leave content untouched.
- README now documents `wet entity rename` usage (commit c249bd6).

Verified locally: `cargo nextest run` for the entity_rename suite is 11/11 passing (up from 10), and `cargo clippy --all-targets` shows only pre-existing unrelated warnings.

Two low-severity nits remain (redundant collision check between the CLI layer and `EntitiesRepository::rename`; error double-printed via `eprintln!` + `main.rs`'s generic handler) but neither blocks — approving.
